projects/plugins/jetpack/class.jetpack.php

```diff
@@ -6896,7 +6896,17 @@ private function plugin_post_activation() {
 		if ( get_transient( 'activated_jetpack' ) ) {
 			delete_transient( 'activated_jetpack' );
 
-			wp_safe_redirect( static::admin_url( 'page=my-jetpack' ) );
+			if ( ( new Host() )->is_woa_site() ) {
+				$redirect_url = static::admin_url( 'page=jetpack' );
+			} elseif ( is_network_admin() ) {
```

Comment:
Do we need this WoA use-case? Since Jetpack is a managed plugin on Atomic, are there ever situations when the plugin is not enabled?

Reply:
I believe you need Jetpack Beta to deactivate Jetpack on WoA, but it's still possible.
So we could get by without it, but I believe having it there is better so site owners wouldn't run into the error after some occasional experiments.